executor/adapter.go

```diff
@@ -167,3 +175,39 @@ func (a *statement) logSlowQuery() {
 		log.Warnf("[%d][TIME_QUERY] %v %s", connID, costTime, sql)
 	}
 }
+
+// isPointGetWithPKOrUniqueKeyByAutoCommit returns true when meets following conditions:
+//  1. ctx is auto commit tagged
+//  2. txn is nil
+//  2. plan is point get by pk or unique key
+func isPointGetWithPKOrUniqueKeyByAutoCommit(ctx context.Context, p plan.Plan) bool {
+	// check auto commit
+	if !ctx.GetSessionVars().IsAutocommit() {
+		return false
+	}
+
+	// check txn
+	if ctx.Txn() != nil {
+		return false
+	}
+
+	// check plan
```

Comment:
The chance of chained projection is very low.
I think just check one level is enough.